source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -18,12 +19,19 @@ namespace Http {
 // See: source/docs/http3_upstream.md
 class AlternateProtocolsCacheImpl : public AlternateProtocolsCache {
 public:
-  explicit AlternateProtocolsCacheImpl(TimeSource& time_source);
+  AlternateProtocolsCacheImpl(TimeSource& time_source, std::unique_ptr<KeyValueStore>&& store);
   ~AlternateProtocolsCacheImpl() override;
 
+  // Note this does not do standards-required normalization. Entries requiring
+  // nomalization will simply not be read from cache.
+  static std::string protocolsToString(const std::vector<AlternateProtocol>& protocols,
+                                       TimeSource& time_source);
+  static absl::optional<std::vector<AlternateProtocol>>
+  protocolsFromString(absl::string_view protocols, TimeSource& time_source,
+                      bool from_cache = false);
```

Comment:
Ah, nice! Much clearer! Reading it now, I wonder if it might be slightly cleaner to rename it to `ma_is_absolute_time` (or some such) since that expresses the semantics a bit more clearly? But this is *super* minor. The new comments definitely make it clear.